vsm

```diff
@@ -146,32 +150,57 @@ def process(state, signal, value):
 
     state.got_signal(signal, value)
 
-
-def run(state):
+def receive():
     '''
-        Loop to grab logic from stdin, then spawn a threads for processing
+        Loop to grab logic from stdin
     '''
     for line in sys.stdin:
         line = line.replace(" ", "").strip('\n')
         if line == "":
             # Ignore blank lines
             pass
         if line == "quit":
-            break
+            exit(0)
         elif "=" in line:
             line = line.split("=")
-            process(state, line[0], line[1])
+            return (line[0], line[1])
         else:
-            process(state, line, None)
+            return (line, None)
 
+def run(state):
+    try:
+        while True:
+            message = receive()
+            if message == None:
+                exit(0)
+
+            signal, value = message
+            # 'quit' signal to close VSM end point.
```

Comment:
Add a line before this comment, and also add a note that this is to simplify testing, please.